src/mono/browser/runtime/diagnostics/diag-ws.ts

```diff
@@ -0,0 +1,61 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+import { IDiagConnection, DiagConnectionBase, diagnostic_server_loop, schedule_diagnostic_server_loop } from "./common";
+
+export function createDiagConnectionWs (socket_handle:number, url:string):IDiagConnection {
+    return new DiagConnectionWS(socket_handle, url);
+}
+
+// this is used together with `dotnet-dsrouter` which will create IPC pipe on your local machine
+// 1. run `dotnet-dsrouter server-websocket` this will print process ID and websocket URL
+// 2. configure your wasm dotnet application `.withEnvironmentVariable("DOTNET_DiagnosticPorts", "ws://127.0.0.1:8088/diagnostics")`
+// 3. run your wasm application
+// 4. run `dotnet-gcdump -p <process ID>` or `dotnet-trace collect -p <process ID>`
+class DiagConnectionWS extends DiagConnectionBase implements IDiagConnection {
+    private ws: WebSocket;
+
+    constructor (client_socket:number, url:string) {
+        super(client_socket);
+        const ws = this.ws = new WebSocket(url);
+        const onMessage = async (evt:MessageEvent<Blob>) => {
+            const buffer = await evt.data.arrayBuffer();
+            const message = new Uint8Array(buffer);
+            this.messagesReceived.push(message);
+            diagnostic_server_loop();
+        };
+        ws.addEventListener("open", () => {
+            for (const data of this.messagesToSend) {
+                ws.send(data);
+            }
+            this.messagesToSend = [];
+            diagnostic_server_loop();
+        }, { once: true });
+        ws.addEventListener("message", onMessage);
+        ws.addEventListener("error", () => {
+            ws.removeEventListener("message", onMessage);
```

Comment:
Is there no error message from the error? Is it not worth it logging it in any way?

Reply:
added warning. It's not worth handling the error because the developer would have to reload page anyway.